solr/core/src/java/org/apache/solr/util/stats/InstrumentedHttpRequestExecutor.java

```diff
@@ -147,6 +152,13 @@ public HttpResponse execute(HttpRequest request, HttpClientConnection conn, Http
     }
   }
 
+  private static final TextMapSetter<HttpRequest> TRACE_INJECTOR = (r, k, v) -> r.setHeader(k, v);
+
+  private static void injectContextIntoRequest(HttpRequest request) {
+    TextMapPropagator textMapPropagator = TraceUtils.getTextMapPropagator();
+    textMapPropagator.inject(Context.current(), request, TRACE_INJECTOR);
+  }
+
```

Comment:
> Is declaring TRACE_INJECTOR useful for a performance perspective? It kinda makes me sad to not see a trivial lambda used as such.

yeah it's coming from their javadocs (both TextMapSetter and TextMapGetter)
```
 * <p>{@code Setter} is stateless and allows to be saved as a constant to avoid runtime allocations.
```

Reply:
> Worthy of it's own JIRA issue IMO.

agreed. I wanted to cover all feedback received on the PR and in doing so, I lost focus a bit. I will break this off into its own Jira.